lib/protokoll/counter.rb

```diff
@@ -30,18 +32,21 @@ def self.build_attrs(object, options)
     end
 
     def self.outdated?(record, options)
-      Time.now.utc.strftime(update_event(options)).to_i > record.updated_at.strftime(update_event(options)).to_i
+      event = update_event(options)
+      return false if event.empty?
+
+      Time.now.utc.strftime(event).to_i > record.updated_at.strftime(event).to_i
     end
 
     def self.update_event(options)
       pattern = options[:pattern]
       event = String.new
 
-      event += "%Y" if pattern.include? "%y" or pattern.include? "%Y"
-      event += "%m" if pattern.include? "%m"
-      event += "%H" if pattern.include? "%H"
-      event += "%M" if pattern.include? "%M"
-      event += "%d" if pattern.include? "%d"
+      event << "%Y" if pattern.include? "%y" or pattern.include? "%Y"
```

Comment:
replacing `+=` with `<<` to avoid an extra string allocation on each time we `+=`

```ruby
irb(main):001:0> x = "foo"
=> "foo"
irb(main):002:0> x.object_id
=> 180
irb(main):003:0> x += "f"
=> "foof"
irb(main):004:0> x.object_id
=> 200
irb(main):005:0> x << "b"
=> "foofb"
irb(main):006:0> x.object_id
=> 200
```